locations/spiders/publix.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+import scrapy
+import re
+
+from locations.items import GeojsonPointItem
+
+
+class PublixSpider(scrapy.Spider):
+    name = "publix"
+    allowed_domains = ['publix.com']
+    start_urls = (
+        'http://www.publix.com/sitemap.xml',
+    )
+
+    def parse(self, response):
+        response.selector.remove_namespaces()
+        city_urls = response.xpath('//url/loc/text()').extract()
+        regex = re.compile(r'http\S+publix.com/locations/\d+\S+')
+        for path in city_urls:
+            if re.search(regex,path):
+                yield scrapy.Request(
+                    path.strip(),
+                    callback=self.parse_store,
+                )
+            else:
+                pass
+
+    def parse_store(self, response):
+
+        if "CLOSED" is response.xpath('//span[@class="store-status"]/text()').extract_first():
+            storeHours = 'STORE CLOSED'
+
+        else:
+            storeHoursHTML = response.xpath('//div[@class="store-info-group"]').extract()[4]
+            p = re.compile(r'<.*?>')
+            storeHours = p.sub('', storeHoursHTML)
+            storeHours = storeHours.replace('\t', '').replace('\r', '').replace('\n', '').replace('       ', ' ')
+            storeHours = "".join(storeHours.strip())
+
+
+        properties = {
+        'name': " ".join(response.xpath('//h1[@id="content_2_TitleTag"]/text()').extract_first().split()),
+        'ref': "".join(response.xpath('//div[@class="store-info-group"]/text()').extract_first().strip()),
```

Comment:
You shouldn't need any of these `"".join()`'s in here. `extract_first()` already returns a string.